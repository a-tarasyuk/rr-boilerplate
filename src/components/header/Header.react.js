import './header.css';
import { Link, IndexLink } from 'react-router';
import React from 'react';

export default () => {
  return <div className="header">
    <IndexLink
      to="/"
      className="header__link"
      activeClassName="header__link_active"
    >
      Home
    </IndexLink>

    <Link
      to="/features"
      className="header__link"
      activeClassName="header__link_active"
    >
      Libraries/Tools
    </Link>
  </div>;
};