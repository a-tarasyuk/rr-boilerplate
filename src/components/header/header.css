.header {
  margin: 10px 0 0;
  padding: 10px 0;
  border-bottom: 1px solid #EEEEEE;

  &__link {
    font-size: 1.8em;
    color: #3572B0;
    display: inline-block;
    margin: 0 10px 0 0;
    text-decoration: none;

    &:hover {
      color: #CCCCCC;
    }

    &_active {
      text-decoration: underline;
    }
  }
}