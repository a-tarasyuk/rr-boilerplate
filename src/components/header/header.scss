@import '~styles/theme/variables';

.header {
  margin: 10px 0 0;
  padding: 10px 0;
  border-bottom: 1px solid $gallery;

  &__link {
    font-size: 1.8em;
    color: $astral;
    display: inline-block;
    margin: 0 10px 0 0;
    text-decoration: none;

    &:hover {
      color: $silver;
    }

    &_active {
      text-decoration: underline;
    }
  }
}