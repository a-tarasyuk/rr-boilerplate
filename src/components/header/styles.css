.header {
  margin: 10px 0 0;
  padding: 10px 0;
  border-bottom: 1px solid #eee;
}
