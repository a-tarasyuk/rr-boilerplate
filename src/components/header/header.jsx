import React from 'react';
import PropTypes from 'prop-types';
import styles from './header.css';

export const Header = ({ children }) => (
  <div className={styles.header}>
    { children }
  </div>
);

Header.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Header;
