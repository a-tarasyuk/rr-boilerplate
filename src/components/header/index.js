import styles from './styles.css';
import { Link, IndexLink } from 'react-router';
import React from 'react';

export default () => {
  return <div className={ styles.header }>
    <IndexLink
      to="/"
      className={ styles.link }
      activeClassName={ styles.link_active }
    >
      Home
    </IndexLink>

    <Link
      to="/features"
      className={ styles.link }
      activeClassName={ styles.link_active }
    >
      Libraries/Tools
    </Link>
  </div>;
};