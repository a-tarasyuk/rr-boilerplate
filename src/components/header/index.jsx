import React from 'react';
import { Link, IndexLink } from 'react-router';
import styles from './styles.css';

export default () => (
  <div className={styles.header}>
    <IndexLink
      to="/"
      className={styles.link}
      activeClassName={styles.link_active}
    >
      Home
    </IndexLink>

    <Link
      to="/features"
      className={styles.link}
      activeClassName={styles.link_active}
    >
      Libraries/Tools
    </Link>
  </div>
);
