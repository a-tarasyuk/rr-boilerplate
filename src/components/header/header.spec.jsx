import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './header';

describe('<Header />', () => {
  it('matches its snapshot', () => {
    const component = shallow(
      <Header>Content</Header>
    );
    expect(component).toMatchSnapshot();
  });
});
