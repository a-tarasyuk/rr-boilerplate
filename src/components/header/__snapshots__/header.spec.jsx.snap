// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches its snapshot 1`] = `
<div
  className="header"
>
  Content
</div>
`;
